### PR TITLE
Bring in api change

### PR DESCRIPTION
**Pull request by theobarberbany**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -266,3 +266,5 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.14.3-0.20230601165947-6ce0bf390ce3 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )
+
+replace github.com/openshift/api => github.com/joelspeed/api v0.0.0-20240424135613-d3739e09825b
```
Comment (JoelSpeed) [inline go.mod:270]: Can you make this openshift api master rather than using a replace please

File: pkg/controller/vsphere/machine_scope_test.go [hunk 2]
```diff
@@ -320,9 +320,9 @@ func TestPatchMachine(t *testing.T) {
 		},
 	}
 	g.Expect(k8sClient.Create(ctx, configNamespace)).To(Succeed())
-	defer func() {
-		g.Expect(k8sClient.Delete(ctx, configNamespace)).To(Succeed())
-	}()
+	// defer func() {
+	// 	g.Expect(k8sClient.Delete(ctx, configNamespace)).To(Succeed())
+	// }()
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/machine_scope_test.go:325]: Huh?

Comment (theobarberbany) [reply pkg/controller/vsphere/machine_scope_test.go:325]: Yea, this fixed the units locally/ unblocked local development on the pausing stuff. Haven't got around to working out what is actually going on! Will take a look

File: go.mod (hunk 1 above)
Comment (theobarberbany) [reply go.mod:270]: But we've not merged the MAPI -> CAPI Migration changes? Or do they not matter for this?

Comment (JoelSpeed) [reply go.mod:270]: They don't matter for now, the idea is just to bring the repo up to master as it is now

File: pkg/controller/vsphere/machine_scope_test.go (hunk 2 above)
Comment (vr4manta) [reply pkg/controller/vsphere/machine_scope_test.go:325]: I was hitting the same thing.  it seems we prevent openshift-config from being deleted.

**Pull request closed without merging**